### DISCUSSION TITLE
[Security] Access decisions with voters

### DISCUSSION THREAD

**Discussion by amigian74** [symfony/symfony#44761] [Q&A]
Hi,

I'm looking for a best practice using voters for access decision. For most cases this work fine for me, but there are some special situations without a "clean" solution.

My voters using the subject (entity) and as attribute only CRUD functions. So I'm running into problems in two scenarios:

1. For the Create function there is no subject to vote against
2. For "global" functions like to show a list of entities there is no single entity too (checkinh all entities would cause major performance issues)

What is your best practice for these situations?

Thx in advance

**Comment by stof** [comment]
I generally have something like `LIST_USERS` voting on a `null` subject, or `LIST_LIKERS` voting on a Post entity (if I display the list of likers for a given post).
And the same works for creating objects. I vote using a `CREATE_*` attribute on the scope in which the action will create the object.
